packages/core/src/transaction.ts

```diff
@@ -126,4 +126,45 @@ export class Transaction {
 
         return null;
     }
+
+    getPublicKeyFromInput(index: number): Buffer | null {
+        const input = this.inputs[index];
+
+        const scriptVector: Buffer[] = [];
+        const buffer = input.script;
+        for (let offset = 0; offset < buffer.byteLength; ) {
+            const sigItemLen = readVarInt(buffer, offset);
+            offset += encodingLength(sigItemLen);
+            scriptVector.push(buffer.subarray(offset, offset + sigItemLen));
+            offset += sigItemLen;
+        }
+
+        // p2pkh
+        if (scriptVector.length === 2) {
+            if (isPubKey(scriptVector[1])) {
+                return scriptVector[1];
+            }
+        }
```

Comment:
Don't use nested `if` when simplification is possible.

```suggestion
        if (scriptVector.length === 2 && isPubKey(scriptVector[1])) {
            return scriptVector[1];
        }
```